### ISSUE TITLE
[Enhancement]: Multi-title handling

### ISSUE THREAD

**Issue by NicholasConrad** [advplyr/audiobookshelf#2626]
### Describe the feature/enhancement

Ability to handle titles for multiple books released as a single AISN for example: "Publishers' Packs", "Box Sets", "Collected Works", etc...

Some audiobooks are released as a collection of related works under a single ASIN. Audiobookshelf already has the ability to set a single audiobook record as multiple installments of a series, however there is no structured way to enumerate the separate titles of those works contained therin.

In this example, there are four separate works collected into a single release called "Thomas Prescott Series Premium". Audiobookshelf is able to correctly parse that this release contains the first four installments of the Thomas Prescott Series: 

![image](https://github.com/advplyr/audiobookshelf/assets/14020277/f9c91ede-14f4-4484-87cc-f4657fde4204)

However the only *title* for the release is not the names of the individual books in the collected release, but rather the title of the collected release itself.

The four titles of the books contained in the collected release are: "Unforeseen", "Gray Matter", "The Afrikaans", and "Show Me". If we try to search against these titles (perhaps not remembering that they are in the database under a collected release) no results are returned:

![image](https://github.com/advplyr/audiobookshelf/assets/14020277/ab36383a-df5d-417e-8020-f05709785413)

Such a feature could also come in handy for enhancing search for works that have been published under multiple titles, for instance Agatha Christie's "And Then There Were None".

**Comment by advplyr** [comment]
I can't think of where these titles would be stored. Adding an additional attribute that is an array of other book titles doesn't seem right. It seems to me like these titles should either be in the description, subtitle, or in the chapter names.

Can you elaborate on how you see this working?

As a side note the series sequence allowing non-numbers like "1-4" may be going away due to sqlite not supporting natural sorting (alpha-numeric)

**Comment by nichwall** [comment]
Just barely beat me to it. I think this is a duplicate of https://github.com/advplyr/audiobookshelf/issues/785

**Comment by advplyr** [comment]
Yeah this can probably be grouped in with #785 even though that feature request is more involved. From what I can tell Audible doesn't handle this use-case either. The publisher can put the titles in the description like the example shared in that other FR https://www.audible.com/pd/Rogues-Audiobook/B00L1GU3WC

If we did add something like this it would need to be manual because there is none of the providers will be able to give us this info.

**Comment by NicholasConrad** [comment]
>I think this is a duplicate of https://github.com/advplyr/audiobookshelf/issues/785

I did read 785 before posting, that request is about adding additional data to chapters. I just want a place I can list an arbitrary number of discreet title strings that will be included in search results, nothing as fancy as jumping to chapter marks. And for instance in the case of "And then there were none" also published as "Ten Little Indians", the chapters are the same, so 785 would not address the issue, but being able to add the alternate title to a field would.

>Can you elaborate on how you see this working?

The same way text entered into the 'narrators' or 'genres' fields becomes a 'pill', and multiple 'pills' can be entered into the field. Maybe there could be a new field called e.g., 'AKA' or 'alternate titles' that has the 'pill' capability, and anything entered in there is also surfaced in search results.

>As a side note the series sequence allowing non-numbers like "1-4" may be going away due to sqlite not supporting natural sorting (alpha-numeric)

This may prove an interesting alternate avenue to solve the issue. If ABS instead of interpreting "Thomas Prescott Series 1-4" created 4 sperate 'series' entries, "Thomas Prescott Series #1", "Thomas Prescott Series #2", and so on, but the series pill edit dialog included an extra optional field for e.g., 'installment title':

![image](https://github.com/advplyr/audiobookshelf/assets/14020277/6835757b-4062-42f6-9317-d9928593ca53)

then search would grab the main title and subtitle fields, as well as anything entered into the series title field. It wouldn't solve the ten little indians problem, but it would be quite an improvement nonetheless. 

>If we do add something like this it would need to be manual because none of the providers will be able to give us this info.

Right now I'm just hand-typing everything into the subtitles field which is pretty messy with more than two or three titles in a single record.

**Comment by manalive** [comment]
Might also consider implementing via #539. 

In a comment, I requested having the ability to specify a volume name since many books are multi-volume or are collections that contain other books. (Bibles, anthologies, and many Audible collections, for example.)

I don't think I'm unique in handling this by adding them as separate 'discs'. 

It would be awesome to have the ability to split them out by disc number, add a volume name (for displaying as a subheader in the GUI), then specify an ASIN and add a checkbox to "display as additional book title" so we get an entry in the main book list with its own metadata. 

If they get their own alias title records, then they get the Series treatment 'for free'. If the disc titles are included in searches, the collection will also show up. The only other thing would be an extra warning or restriction on writing metadata from the 'single' titles to the files since that could break the collection metadata on a rescan.

I've been copying out the individual books from collections as a workaround to get this functionality. Would love to save time and storage.

Example:
https://www.audible.com/pd/The-Chronicles-of-Narnia-Complete-Audio-Collection-Audiobook/0061999881
is a combination of 
https://www.audible.com/pd/The-Lion-the-Witch-and-the-Wardrobe-Audiobook/B002V1NHJ2
https://www.audible.com/pd/Prince-Caspian-Audiobook/B002V8OEW4
https://www.audible.com/pd/The-Voyage-of-the-Dawn-Treader-Audiobook/B002V1LIK2
https://www.audible.com/pd/The-Silver-Chair-Audiobook/B002VA8HZ2
https://www.audible.com/pd/The-Horse-and-His-Boy-Audiobook/B002V57SW0
https://www.audible.com/pd/The-Magicians-Nephew-Audiobook/B002V0JTCW
https://www.audible.com/pd/The-Last-Battle-Audiobook/B002UZJF22

Here are more examples of audible books made up of other audible books:
https://www.audible.com/pd/C-S-Lewis-Audiobook/B00HJZDKLU
https://www.audible.com/pd/C-S-Lewis-Essential-Audio-Library-Audiobook/0063136686
https://www.audible.com/pd/The-Adventure-Collection-Audiobook/B00G4A9HVS
https://www.audible.com/pd/The-Jane-Austen-Collection-Audiobook/B08JZF5T49
https://www.audible.com/pd/Thornton-W-Burgess-Box-Set-Audiobook/B08QPXHY56
https://www.audible.com/pd/The-Roald-Dahl-Audio-Collection-Audiobook/B00BWUBLFU

----

> If we do add something like this it would need to be manual because none of the providers will be able to give us this info.

The providers are AWFUL. Audible is the biggest source of data but it is terrible. All of us who take metadata seriously do a ton of manual management. ABS can become the standard by creating a place to put data the providers don't supply. It's probably the most popular solution at this point, isn't it?
